libs/zzz/db/src/Database/DataManagers/CharacterOptManager.ts

```diff
@@ -140,11 +243,39 @@ export class CharacterOptManager extends DataManager<
       })
       .filter(notEmpty)
     if (!Array.isArray(bonusStats)) bonusStats = []
-    bonusStats = bonusStats.filter(({ tag, value }) => {
-      if (!validateTag(tag)) return false
-      if (typeof value !== 'number') return false
-      return true
-    })
+    bonusStats = bonusStats.map(
+      ({ tag: { q, qt, attribute, damageType1, damageType2 }, value }) => {
+        if (typeof value !== 'number') value = 0
+        q = validateValue(q, bonusStatKeys) ?? 'atk'
+        qt = validateValue(qt, bonusStatQtKeys) ?? 'combat'
+
+        if (q !== 'dmg_') attribute = undefined
+        if (attribute) attribute = validateValue(attribute, allAttributeKeys)
+
+        if (
+          !bonusStatDmgTypeIncStats.includes(
+            q as (typeof bonusStatDmgTypeIncStats)[number]
+          )
+        )
+          damageType1 = undefined
+        if (damageType1)
+          damageType1 = validateValue(damageType1, bonusStatDamageTypes)
+
+        if (q !== 'dmg_' && q !== 'crit_dmg_') damageType2 = undefined
```

Comment:
why?

Reply:
dt2 is only 'aftershock' and we only have buffs that increase its dmg and critdmg_. I'll put a comment.